### PR TITLE
Fix positional metavar and Optional[str] defaults in clickify

### LINKED ISSUE #415: check passing of Optional[str] defaults from CLI
Not working right with ``breifast tofits``.

### LINKED ISSUE #424: set metavar of positionals in clickify_parameters to name
Metavar is currently set from dtype, which is OK for options, but opaque for positionals. Make default metavar of positionals equal to name.

### PR DESCRIPTION
**Pull request by gijzelaerr**
## Summary

- **Positional metavar** (fixes caracal-pipeline/stimela#424): Set the default metavar of positional Click arguments to the parameter name (uppercased) instead of the dtype string. This makes CLI help text like `Usage: cmd [OPTIONS] INPUT-FILE OUTPUT-DIR` instead of `Usage: cmd [OPTIONS] str str`. The explicit `schema.metavar` field is still respected when set by the user.

- **Optional[str] defaults** (fixes caracal-pipeline/stimela#415): For `Optional[...]` types with unset defaults, explicitly pass `default=None` to Click rather than relying on Click's implicit `None` fallback. This ensures consistent behavior for Optional parameters across all Click configurations and callback/validator scenarios.

## Test plan

- [x] Verified positional metavar shows parameter name in help text
- [x] Verified `schema.metavar` override still works for positionals
- [x] Verified `Optional[str]` with string default passes through correctly
- [x] Verified `Optional[str]` with no default gets explicit `None`
- [x] Verified `Optional[str]` with `null` default gets `None`
- [x] All 59 stimela tests pass (one pre-existing failure in `test_test_recipe` unrelated to this change)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_clickify.py [hunk 1]
```diff
@@ -90,6 +90,150 @@ def test_boolean_policies_explicit_yes_no():
...
+    runner = CliRunner()
+    result = runner.invoke(positional_app, ["--help"])
+    assert result.exit_code == 0
+    # The usage line should not contain bare 'str' as positional metavar.
+    # Options like --count may show 'int' in their help, but the usage line
+    # for positionals (before Options:) should use names.
+    usage_line = result.output.split("\n")[0]
+    # The usage line should not have 'str' as a standalone positional metavar
+    assert "INPUT-FILE" in usage_line
+    assert "OUTPUT-DIR" in usage_line
```
[the remark this answers is not in the document]
Comment (gijzelaerr) [reply tests/test_clickify.py:135]: Valid point. Fixing now — adding an explicit assertion that the dtype string (e.g. 'str') does NOT appear as a standalone metavar in the usage line.

File: tests/test_clickify.py [hunk 2]
```diff
@@ -90,6 +90,150 @@ def test_boolean_policies_explicit_yes_no():
...
+
```
[the remark this answers is not in the document]
Comment (gijzelaerr) [reply tests/test_clickify.py:138]: Same fix as above — adding the missing negative assertion to make this test meaningful rather than a duplicate.

**Review by o-smirnov — APPROVED** [review]
LGTM!

File: tests/test_clickify.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gijzelaerr) [reply tests/test_clickify.py:135]: Fixed — added explicit assertion that 'str' does not appear as a standalone positional metavar in the usage line.

File: tests/test_clickify.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gijzelaerr) [reply tests/test_clickify.py:138]: Fixed — same as above, added negative assertion to make this test meaningful.
